blog/2015-02-26-Getting-Started-With_Docker-And-Wildfly.adoc

```diff
@@ -0,0 +1,195 @@
+
+....
+[root@localhost docker]$ docker ps
+CONTAINER ID        IMAGE                  COMMAND                CREATED             STATUS              PORTS                                            NAMES
+f70149043400        jboss/wildfly:latest   "/opt/jboss/wildfly/   58 seconds ago      Up 58 seconds       0.0.0.0:8080->8080/tcp, 0.0.0.0:9990->9990/tcp   ecstatic_darwin     
+
+[root@localhost docker]$ docker inspect -f '{{ .NetworkSettings.IPAddress }}' f70149043400
+172.17.0.25
+....
+
+To see the Wildfly welcome page, we could open up any browser and browse to `http://172.17.0.25:8080`.  
+But again, this IP address will change every launch.  To get this to work in tools, 
+we'd need to constantly change our host in our 
+Server Editor, which would quickly become tedious. These commands to get the IP address are also pretty tedious and verbose
+and not at all convenient.  
+
+To make this easier, we can map 
+our container's ports to our localhost ports, exposing those from our container as if they were on localhost. 
+
+ docker run -it -p 8080:8080 -p 9990:9990 jboss/wildfly
+
+In this way, we can now open our browser to `http://localhost:8080` to see the Wildfly welcome page. 
+
+=== Mapping a deployment folder
+
+Since we're not exposing SSH in this example, and we're not using the Management profiles in JBossTools, 
+we need a way for our server to pick up our deployments. The easiest way to do this is to map in a local folder 
+from our host into our container. For this example, we'll make a temporary directory somewhere on our host, 
+and tell our Docker container to treat that as the standalone/deployments folder inside the container. In this way, 
+changes the container makes to the folder can be visible on the host, and vice versa. 
+
+*Mapping Folders may cause IO errors for SELinux!*  To ensure your container can actually read and write to the folder, 
+you'll need to run `setenforce 0` to disable SELinux, or, alternatively, give Docker the permissions and exceptions in SELinux. 
+
+....
+[root@localhost jbds_wf_external]$  mkdir /home/rob/tmp/dockertest1
+[root@localhost jbds_wf_external]$  docker run -it -p 8080:8080 -p 9990:9990 \
+-v /home/rob/tmp/dockertest1:/opt/jboss/wildfly/standalone/deployments/:rw  jboss/wildfly
+....
+
+If you were to now place a .war file inside `/home/rob/tmp/dockertest1`, it will be picked up by the deployment scanner, 
+and visible in a web browser. 
+
+=== Making your Server Adapter
+
+The final step of this example is to create your Wildfly 8.2 server adapter in JBossTools, 
+and to create and deploy a web application to your temporary folder, which in my case is
+`/home/rob/tmp/dockertest1`
+
+First, we'll open the `Servers View` and create a new `Wildfly 8.x` server adapter.
+Since we're exposing our container's ports on our localhost, we can leave the host as 
+`localhost`. Make sure your server is marked as `Local` and Controlled by `Filesystem and shell operations` 
```

Comment:
no it wont. 

osx maps the ports to the dockerhost which is a unique IP not mapped into localhost (at least not on default boot2docker)